components/style/values.rs

```diff
@@ -1317,6 +1319,17 @@ pub mod specified {
         }
     }
 
+    #[derive(PartialEq, Clone, Debug)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub struct UrlExtraData {
```

Comment:
When supporting background-image, I found it simpler to have `values::specified::UrlExtraData` be defined for both stylo and non-stylo builds, so that types for specified values that need to store away the UrlExtraData don't need to have different shapes.  Otherwise we end up branching with `cfg`s later when doing pattern matching on say `specified::Image` values, since one would have an extra_data field and the other wouldn't.